Userland/Services/WebContent/RenderLoopExecutor.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/EventLoop.h>
+#include <LibGfx/ShareableBitmap.h>
+#include <LibWeb/Painting/PaintingCommandExecutorCPU.h>
+#include <WebContent/RenderLoopExecutor.h>
+
+#ifdef HAS_ACCELERATED_GRAPHICS
+#    include <LibWeb/Painting/PaintingCommandExecutorGPU.h>
+#endif
+
+namespace WebContent {
+
+RenderLoopExecutor::RenderLoopExecutor(bool use_gpu_painter)
+    : m_main_thread_event_loop(Core::EventLoop::current())
```

Comment:
this assumes that we always create this object from the main thread. Is that always true? Will that always be true? Should we pass in the 'owner event loop' as a pointer/reference instead?

Reply:
`RenderLoopExecutor` is always constructed in main thread and `m_main_thread_event_loop` is a reference. or am I missing something?

Reply:
My question is, is this a property we should encode in the Executor's constructor, or should we let the caller make that explicit EventLoop::current() method themselves. It's also possible that the current event loop is not the main thread event loop, so the name is a bit misleading.

Reply:
got it, yes, I renamed it to `m_owner_thread_event_loop` which should be less confusing name.